domain/model/service/service.go

```diff
@@ -151,6 +162,42 @@ func NewService(
 	}
 }
 
+// WatcherFactory describes methods for creating watchers.
+type WatcherFactory interface {
+	// NewNamespaceWatcher returns a new namespace watcher for events based on the input change mask.
+	// The initialStateQuery ensures the watcher starts with the current state of the table,
```

Comment:
I would change table here to system. There is no where we say this must be restricted to a table. That is an implementation detail we want to avoid suggesting.